pyrometheus/__init__.py

```diff
@@ -362,17 +474,102 @@ def get_temperature(self, H_or_E, T_guess, Y, do_energy=False):
 
         return T
 
+    def get_falloff_rates(self, T, C, k_fwd):
+        k_high = np.array([
+        %for react in falloff_reactions:
+            ${cgm(rate_coefficient_expr(react.high_rate, Variable("T")))},
+        %endfor
+        ])
+
+        k_low = np.array([
+        %for react in falloff_reactions:
+            ${cgm(rate_coefficient_expr(react.low_rate, Variable("T")))},
+        %endfor
+        ])
+
+        reduced_pressure = np.array([
+        %for i, react in enumerate(falloff_reactions):
+            (${cgm(third_body_efficiencies_expr(
+                sol, react, Variable("C")))})*k_low[${i}]/k_high[${i}],
+        %endfor
+        ])
+
+        falloff_center = np.array([
+        %for react in falloff_reactions:
+            %if react.falloff.falloff_type == "Troe":
+            np.log10(${cgm(troe_falloff_expr(react, Variable("T")))}),
+            %else:
+            1,
+            %endif
+        %endfor
+        ])
+
+        falloff_function = np.array([
+        %for i, react in enumerate(falloff_reactions):
+            ${cgm(falloff_function_expr(
+                react, i, Variable("T"), Variable("reduced_pressure"),
+                Variable("falloff_center")))},
+        %endfor
+        ])*reduced_pressure/(1+reduced_pressure)
+
+        %for i, react in enumerate(falloff_reactions):
+        k_fwd[${int(react.ID)-1}] = k_high[${i}]*falloff_function[${i}]
+        %endfor
+        return
+
+    def get_fwd_rate_coefficients(self, T, C):
+        k_fwd = np.array([
+        %for react in sol.reactions():
+        %if isinstance(react, ct.FalloffReaction):
+            0,
+        %else:
+            ${cgm(rate_coefficient_expr(react.rate, Variable("T")))},
+        %endif
+        %endfor
+        ])
+        %if falloff_reactions:
+        self.get_falloff_rates(T, C, k_fwd)
+        %endif
+
+        %for react in three_body_reactions:
+        k_fwd[${int(react.ID)-1}] *= (${cgm(third_body_efficiencies_expr(
```

Comment:
Wouldn't the index of `react` match `int(react.ID)-1`?

Reply:
Analogous to the above, though for three-body reactions.

Reply:
OIC. Thanks for explaining!